### PR TITLE
Update Gemma and Gemma 2 to more closely follow Python implementations

### PR DESCRIPTION
**Pull request by DePasqualeOrg**
I've been having problems with Gemma 2 in particular, so I made these adjustments to see if they help. A few small implementation details have changed, which I'll mark with comments. I've verified that this works with Gemma 2B and Gemma 2 2B.

Some of these changes are merely cosmetic, but maybe keeping the Swift implementation as close as possible to the Python implementation will help with in-context learning when using LLMs to port new models to Swift, and with training new LLMs that will be used to produce MLX code in Swift.

### CODE REVIEW

File: Libraries/LLM/Models/Gemma.swift [hunk 1]
```diff
@@ -171,29 +177,29 @@ public class GemmaModelInner: Module {
...
         self.headDim = .init(args.headDimensions)
         self.model = GemmaModelInner(args)
     }
 
     public func callAsFunction(_ inputs: MLXArray, cache: [KVCache]?) -> MLXArray {
-        var out = model(inputs, cache: cache)
-        out = matmul(out, model.embedTokens.weight.T)
-        return out
+        let out = model(inputs, cache: cache)
+        return model.embedTokens.asLinear(out)
```
Comment (DePasqualeOrg) [inline Libraries/LLM/Models/Gemma.swift:197]: This has changed.

File: Libraries/LLM/Models/Gemma2.swift [hunk 2]
```diff
@@ -237,7 +229,7 @@ public struct Gemma2Configuration: Codable {
     var ropeTraditional: Bool = false
     var attnLogitSoftcapping: Float = 50.0
     var finalLogitSoftcapping: Float = 30.0
-    var queryPreAttnScalar: Int = 256
+    var queryPreAttnScalar: Float = 144.0
```
Comment (DePasqualeOrg) [inline Libraries/LLM/Models/Gemma2.swift:232]: This has changed.

File: Libraries/LLM/Models/Gemma2.swift [hunk 3]
```diff
@@ -81,33 +79,31 @@ private class Attention: Module {
             keys = rope(keys)
         }
 
-        let repeats = self.args.attentionHeads / self.args.kvHeads
+        queries = queries * self.scale
```
Comment (DePasqualeOrg) [inline Libraries/LLM/Models/Gemma2.swift:82]: This has changed.

**Review by awni — APPROVED** [review]
Great changes, thanks for the fixes!!

**Pull request merged**